scale-tests/streaming_test.py

```diff
@@ -1,102 +1,115 @@
-                       "--conf", "spark.executor.cores=2",
-                       "--class", "KafkaFeeder"] + args
+        spark_utils.submit_job(app_url=jar_url,
+                               app_args=consumer_args,
+                               app_name=spark_app_name,
+                               args=consumer_config,
+                               driver_role=driver_role,
+                               verbose=False)
 
-    utils.submit_job(app_url=JAR_URI,
-                     app_args=producer_args,
-                     app_name=spark_app_name,
-                     args=producer_config,
-                     verbose=False)
+    common_conf = [
+        "--supervise",
+        "--conf", "spark.mesos.containerizer=mesos",
+        "--conf", "spark.mesos.driver.failoverTimeout=30",
+        "--conf", "spark.mesos.executor.docker.image=mesosphere/spark-dev:f5dd540adffd9ab9e3e826e48d22e39ebc296567-1d7926a8b500d0105b80a6bb808a671b047dc963",
+        "--conf", "spark.mesos.uris=http://norvig.com/big.txt",
+        "--conf", "spark.port.maxRetries=32",
+        "--conf", "spark.scheduler.maxRegisteredResourcesWaitingTime=2400s",
+        "--conf", "spark.scheduler.minRegisteredResourcesRatio=1.0"
+    ]
 
+    broker_dns = _kafka_broker_dns()
+    num_words_to_read = int(desired_runtime * PRODUCER_NUM_WORDS_PER_MIN)
 
-def _submit_consumers(broker_dns, args, topic, spark_app_name, num_words):
-    consumer_args = " ".join([broker_dns, topic, num_words, KERBEROS_FLAG])
+    for dispatcher in dispatchers:
+        spark_app_name, driver_role = dispatcher.split(",")
+        topic = "topic-{}".format(spark_app_name)
 
-    consumer_config = ["--conf", "spark.cores.max=4",
-                       "--class", "KafkaConsumer"] + args
+        _submit_producer(broker_dns, common_conf, topic, spark_app_name, driver_role)
 
-    utils.submit_job(app_url=JAR_URI,
-                     app_args=consumer_args,
-                     app_name=spark_app_name,
-                     args=consumer_config,
-                     verbose=False)
+        for _ in range(0, num_consumers):
+            _submit_consumer(broker_dns, common_conf, topic, spark_app_name, driver_role, str(num_words_to_read))
 
 
 if __name__ == "__main__":
-    usage = """
-        Setup: export PYTHONPATH=../spark-testing:../testing:../tests
-        Usage: python streaming_test.py [dispatcher_file] [num_consumers_per_producer] [desired_runtime_in_mins]
-    """
-
-    if len(sys.argv) < 4:
-        print(usage)
-        sys.exit(2)
-
-    dispatchers_file = sys.argv[1]
-    print("dispatchers_file: {}".format(dispatchers_file))
-    with open(dispatchers_file) as f:
-        dispatchers = f.read().splitlines()
-    print("dispatchers: {}".format(dispatchers))
+    args = docopt(__doc__)
 
-    num_consumers = int(sys.argv[2])
-    print("num_consumers_per_producer: {}".format(num_consumers))
+    with open(args["<dispatcher_file>"]) as f:
```

Comment:
Note for myself: I wasn't expecting the `<` and `>` to be needed.

Reply:
Seems docopt just takes the entire string as a param just like how the dashes are required for arg `["--jar"]`.